platform/security/secure-boot/src/main/java/org/codice/ddf/security/boot/SecureBoot.java

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.security.boot;
+
+import com.google.common.annotations.VisibleForTesting;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import org.apache.karaf.system.SystemService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SecureBoot {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(SecureBoot.class);
+
+  private static final String DDF_HOME_SYS_PROP = "ddf.home";
+
+  private static final String USER_HOME_SYS_PROP = "user.home";
+
+  private static final int INSECURE_BOOT_EXIT_CODE = 15;
+
+  private static final String NOW = "0";
+
+  private final SystemService systemService;
+
+  private Path ddfHome;
+
+  private Path userHome;
+
+  private boolean shuttingDown = false;
+
+  public SecureBoot(SystemService systemService) {
+    this.systemService = systemService;
+    try {
+      ddfHome = getDdfHome();
+      userHome = getUserHome();
+    } catch (IOException e) {
+      String message =
+          "Unable to determine if your installation ["
+              + ddfHome
+              + "] is located in an insecure directory. See log for details. Shutting down...";
+      System.err.println("\n" + message + "\n");
+      LOGGER.error(message, e);
```

Comment:
❓ Do we want to log exceptions and their stack trace at the ERROR level? Should the stack trace be done at DEBUG instead?

Reply:
fixed